### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Dear!

**We implemented the following requirements.**

- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and 
  named accordingly (e.g. rentals.json)
- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -93,6 +104,17 @@ def create_a_rental
...
+    jsonarray = []
+    @rental.each do |item|
+      jsonarray.push({ date: item.date,
+                       book: { title: item.book.title, author: item.book.author },
+                       person: { id: item.person.id, name: item.person.name, age: item.person.age } })
+    end
+    File.write('rental.json', JSON.generate(jsonarray))
   end
 
   def list_all_rentals
```
Comment (brytebee) [inline app.rb:132]: - [ ] **Required:** I observed that when a rental is created in a session of the app and the session is exited, the rentals is preserved. However, on reload, only the rental is not re-read into the app. Kindly check this out. [Check this for help.](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/store_data_in_files.md#storing-data-in-files)

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @SaboorSohaib & @Ngufor-emmanuella👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great console-UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ App runs smoothly

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
- [ ] Like the previous reviewer mentioned, the rentals error still persists. Please ensure that all rentals are successfully shown after the app has been exited and started again.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._
- I suggest you remove the ```Microverse README template``` writing from your README and replace it with the name of your project as this makes your README more professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SaboorSohaib** [comment]
Hi @Ghiftee 
Thank you for reviewing I tried to figure out the issues but I couldn't. Could you please give me a hint?

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @SaboorSohaib , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**